src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php

```diff
@@ -76,5 +80,21 @@ protected function initialize()
                 $this->addLoader($alias, $this->container->get($id));
             }
         }
+
+        if ($this->resourceDirs) {
+            $finder = Finder::create()
+                ->files()
+                ->filter(function (\SplFileInfo $file) {
+                    return 2 === substr_count($file->getBasename(), '.') && preg_match('/\.\w+$/', $file->getBasename());
+                })
+                ->in($this->resourceDirs)
+            ;
```

Comment:
Instead of passing the resource directories, what about passing the list of files and move back this piece of code in the extension? That would make things faster and safe as if the list of files changes, the container is going to be rebuilt anyway.